CMakeLists.txt

```diff
@@ -3,269 +3,514 @@
+#set(CMAKE_VERBOSE_MAKEFILE 1)
+
+
+# -----------------------------------------------------------------------------
+# Set compilation options according to build type
+
+IF("${CMAKE_BUILD_TYPE}" MATCHES "Debug")
+    message(STATUS "building for: debugging")
+
+    ## unfortunately these produce errors
+    #include(CheckCXXCompilerFlag)
+    #CHECK_CXX_COMPILER_FLAG("-Wformat-signedness" CXX_FORMAT_SIGNEDNESS)
+    #CHECK_CXX_COMPILER_FLAG("-Werror=format-security" CXX_FORMAT_SECURITY)
+    #CHECK_CXX_COMPILER_FLAG("-fstack-protector-all" CXX_STACK_PROTECTOR)
+    set(CXX_FORMAT_SIGNEDNESS "-Wformat-signedness")
+    set(CXX_FORMAT_SECURITY "-Werror=format-security")
+    set(CXX_STACK_PROTECTOR "-fstack-protector-all")
+
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O1 -ggdb -Wall -Wextra -DNETDATA_INTERNAL_CHECKS=1 -DNETDATA_VERIFY_LOCKS=1 ${CXX_FORMAT_SIGNEDNESS} ${CXX_FORMAT_SECURITY} ${CXX_STACK_PROTECTOR}")
+ELSE()
+    message(STATUS "building for: release")
+    cmake_policy(SET CMP0069 "NEW")
+    include(CheckIPOSupported)
+    check_ipo_supported(RESULT ipo_supported OUTPUT error)
+    IF(${ipo_supported})
+        message(STATUS "link time optimization: supported")
+        set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
+    ELSE()
+        message(STATUS "link time optimization: not supported")
+    ENDIF()
+ENDIF()
+
+
+# -----------------------------------------------------------------------------
+# O/S Detection
+
+# these are defined in common.h too
+SET(LINUX   False)
+SET(FREEBSD False)
+SET(MACOS   False)
+
+# Detect the operating system
+IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+    SET(TARGET_OS_NAME "macos")
+    SET(TARGET_OS 3)
+    SET(MACOS True)
+ELSEIF(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
+    SET(TARGET_OS_NAME "freebsd")
+    SET(TARGET_OS 2)
+    SET(FREEBSD True)
+ELSE()
+    SET(TARGET_OS_NAME "linux")
+    SET(TARGET_OS 1)
+    SET(LINUX True)
+ENDIF()
+
+# show the operating system on the console
+message(STATUS "operating system: ${TARGET_OS_NAME} (TARGET_OS=${TARGET_OS})")
+
+
+# -----------------------------------------------------------------------------
+# Detect libuuid
+
+pkg_check_modules(UUID REQUIRED uuid)
+set(NETDATA_COMMON_CFLAGS ${NETDATA_COMMON_CFLAGS} ${UUID_CFLAGS_OTHER})
+set(NETDATA_COMMON_LIBRARIES ${NETDATA_COMMON_LIBRARIES} ${UUID_LIBRARIES})
+set(NETDATA_COMMON_INCLUDE_DIRS ${NETDATA_COMMON_INCLUDE_DIRS} ${UUID_INCLUDE_DIRS})
+
+# -----------------------------------------------------------------------------
+# Detect libz
+
+pkg_check_modules(ZLIB REQUIRED zlib)
+set(NETDATA_COMMON_CFLAGS ${NETDATA_COMMON_CFLAGS} ${ZLIB_CFLAGS_OTHER})
+set(NETDATA_COMMON_LIBRARIES ${NETDATA_COMMON_LIBRARIES} ${ZLIB_LIBRARIES})
+set(NETDATA_COMMON_INCLUDE_DIRS ${NETDATA_COMMON_INCLUDE_DIRS} ${ZLIB_INCLUDE_DIRS})
+
+
+# -----------------------------------------------------------------------------
+# Detect libcap
+
+IF(LINUX)
+    pkg_check_modules(CAP QUIET libcap)
```

Comment:
FreeBSD has built-in support for [Capabilities](https://www.freebsd.org/cgi/man.cgi?query=capsicum&sektion=4&apropos=0&manpath=FreeBSD+11.2-RELEASE+and+Ports) and [POSIX.1e](https://www.freebsd.org/cgi/man.cgi?query=posix1e&apropos=0&sektion=0&manpath=FreeBSD+11.2-RELEASE+and+Ports&arch=default&format=html)

Reply:
Is that compatible with `libcap` under linux? `apps.plugin` uses Linux capabilities instead of `setuid` to root.

Reply:
No. The API is different.

Reply:
So, we don't support it yet.